### PR TITLE
Improve cleaning config handling

### PR DESCRIPTION
**Pull request by maxnoe**

### CODE REVIEW

**Comment by moralejo** [comment]
Just one comment (not really on the change, with which I agree): perhaps we should move to main_island being false by default.

**Comment by maxnoe** [comment]
Another question that popped up when looking at this is that I would change the order.

I would calculate number of islands / main island *after* applying the time delta cleaning.

**Comment by maxnoe** [comment]
But that can be another PR, this is uncontroversial, right? So lets's go ahead with this and I'll open more PRs for the other stuff.

**Review by mexanick — APPROVED** [review]

**Comment by rlopezcoto** [comment]
is this again introducing the bug that was solved in #637?

**Comment by maxnoe** [comment]
No

File: lstchain/reco/r0_to_dl1.py [hunk 1]
```diff
@@ -103,22 +103,12 @@ def get_dl1(
...
-    if "delta_time" in cleaning_parameters:
-        delta_time = cleaning_parameters["delta_time"]
-
-    # we use pop because ctapipe won't recognize that keyword in tailcuts
-    cleaning_parameters_for_tailcuts.pop("delta_time")
-    cleaning_parameters_for_tailcuts.pop("use_only_main_island")
-    
+
+    # pop delta_time and use_main_island, so we can cleaning_parameters to tailcuts
+    cleaning_parameters = config["tailcut"].copy()
```
Comment (maxnoe) [inline lstchain/reco/r0_to_dl1.py:108]: The copy is still here @rlopezcoto

Comment (rlopezcoto) [reply lstchain/reco/r0_to_dl1.py:108]: ah ok, I missed that, thanks (test were passing, so it meant that changes were fine anyways)

**Review by rlopezcoto — APPROVED** [review]

**Review by rlopezcoto — APPROVED** [review]

**Pull request merged**